plugins/home/src/api/VisitsApiFactory.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Visit,
+  VisitsApi,
+  VisitsApiQueryParams,
+  VisitsApiSaveParams,
+} from './VisitsApi';
+
+type ArrayElement<A> = A extends readonly (infer T)[] ? T : never;
+
+/** @public */
+export type VisitsApiFactoryOptions = {
+  randomUUID: Window['crypto']['randomUUID'];
```

Comment:
Nit: I guess that window.crypto is not exactly uncommon these days, but from a type perspective, this might as well be `() => string` to keep it clean. Feels odd to have `Window` in the types for some reason 😅 Same in other places